File: backend/app/schemas/analytics.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class PostEngagementResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    post_id: str
    content_preview: str
    page_name: str
    likes: int
    comments: int
    shares: int
    published_at: Optional[datetime] = None


class EngagementListResponse(BaseModel):
    posts: list[PostEngagementResponse]


class HeatmapCell(BaseModel):
    day: int = Field(ge=0, le=6)
    hour: int = Field(ge=0, le=23)
    avg_engagement: float


class HeatmapResponse(BaseModel):
    cells: list[HeatmapCell]

File: frontend/src/types/analytics.ts
export interface PostEngagement {
  post_id: string
  content_preview: string
  page_name: string
  likes: number
  comments: number
  shares: number
  published_at?: string
}

export interface HeatmapCell {
  day: number
  hour: number
  avg_engagement: number
}
